src/DevTools.php

```diff
@@ -277,4 +297,113 @@ private function buildPhar(CommandSender $sender, string $pharPath, string $base
 			$sender->sendMessage("[DevTools] $line");
 		}
 	}
+
+	private function describeHandlerList(CommandSender $sender, HandlerList $handlerList, string $className) : bool{
+		$found = false;
+		foreach(EventPriority::ALL as $priority){
+			$priorityName = self::EVENT_PRIORITY_NAMES[$priority];
+			for($currentList = $handlerList; $currentList !== null; $currentList = $currentList->getParent()){
+				$handlers = $handlerList->getListenersByPriority($priority);
+				if(count($handlers) === 0){
+					continue;
+				}
+
+				if(!$found){
+					$found = true;
+					$sender->sendMessage("--- Handlers called by " . TextFormat::GREEN . $className . TextFormat::WHITE . " ---");
+				}
+
+				foreach($handlers as $handler){
+					$sender->sendMessage(
+						"- " .
+						TextFormat::DARK_GREEN . Utils::getNiceClosureName($handler->getHandler()) . TextFormat::RESET .
+						" in plugin " .
+						TextFormat::DARK_GREEN . $handler->getPlugin()->getName() . TextFormat::RESET .
+						" at priority " .
+						TextFormat::DARK_GREEN . $priorityName . TextFormat::RESET .
+						" (handles cancelled events: " .
+						TextFormat::DARK_GREEN . ($handler->isHandlingCancelled() ? "yes" : "no") . TextFormat::RESET .
+						")"
+					);
+				}
+			}
+		}
+
+		return $found;
+	}
+
+	/**
+	 * @param string[] $args
+	 */
+	private function handlerListCommand(CommandSender $sender, array $args) : bool{
+		if(count($args) === 0){
+			$all = HandlerListManager::global()->getAll();
+			ksort($all, SORT_STRING);
+			foreach($all as $className => $handlerList){
+				$this->describeHandlerList($sender, $handlerList, $className);
+			}
+			return true;
+		}elseif(count($args) > 1){
+			return false;
+		}
+
+		//getListFor() will throw if the class is not valid, but the validation conditions aren't exposed to the API,
+		//so this is the best way to handle missing classes.
+		$handlerList = HandlerListManager::global()->getAll()[$args[0]] ?? null;
```

Comment:
can we try to use substring match here?

Reply:
Yeah, I suppose that would be nice, but it won't be fast.